superset/views/users/api.py

```diff
@@ -93,3 +97,85 @@ def get_my_roles(self) -> Response:
             return self.response_401()
         user = bootstrap_user_data(g.user, include_perms=True)
         return self.response(200, result=user)
+
+
+class UserRestApi(BaseSupersetApi):
+    """An API to get information about users"""
+
+    resource_name = "user"
+    openapi_spec_tag = "User"
+    openapi_spec_component_schemas = (UserResponseSchema,)
+
+    @expose("/<user_id>/avatar.png", methods=("GET",))
+    @safe
+    def avatar(self, user_id: str) -> Response:
+        """Get a redirect to the avatar's URL for the user with the given ID.
+        ---
+        get:
+          summary: Get the user avatar
+          description: >-
+            Gets the avatar URL for the user with the given ID, or returns a 401 error
+            if the user is unauthenticated.
+          parameters:
+            - in: path
+              name: user_id
+              required: true
+              description: The ID of the user
+              schema:
+                type: string
+          responses:
+            301:
+              description: A redirect to the user's avatar URL
+            401:
+              $ref: '#/components/responses/401'
+            404:
+              $ref: '#/components/responses/404'
+        """
+        try:
+            avatar_url = None
+
+            if not user_id.isdigit():
+                return self.response_404()
+
+            user_id_as_int: int = int(user_id)
+
+            user = (
+                db.session.query(security_manager.user_model)
+                .filter_by(id=user_id_as_int)
+                .first()
+            )
+            if not user:
+                return self.response_404()
+
+            # fetch from the one-to-one relationship
+            if len(user.extra_attributes) > 0:
+                avatar_url = user.extra_attributes[0].avatar_url
+
+            should_fetch_slack_avatar = app.config.get(
+                "SLACK_ENABLE_AVATARS"
+            ) and app.config.get("SLACK_API_TOKEN")
+            if not avatar_url and should_fetch_slack_avatar:
+                # Fetching the avatar url from slack
+                avatar_url = get_user_avatar(user.email)
+
+                # Saving the avatar url to the database
+                user_attrs = (
+                    # already exists
+                    db.session.query(UserAttribute)
+                    .filter_by(user_id=user_id_as_int)
+                    .first()
+                    or
+                    # create new
+                    UserAttribute(user_id=user_id_as_int)
+                )
+                user_attrs.avatar_url = avatar_url
+                db.session.add(user_attrs)
+                db.session.commit()
```

Comment:
This is the only element I did not fully address because it's effectively the feature here. Note that mutations will be very very sparse here. Only if:
- feature is on (SLACK_ENABLE_AVATAR + SLACK_API_TOKEN)
- it's the **very first time** the avatar is requested for a given target user in the whole environment, updating the UserAttributes.avatar_url once with a short string

In all normal day-to-day
- upon a new browser requesting an avatar, the GET does as a normal get (a one cell database lookup)
- 301 gets cached, target image gets cached